channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -539,7 +539,9 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
      */
     private _xhrSender(payload: string[], isAsync: boolean) {
         const xhr = new XMLHttpRequest();
-        xhr[DisabledPropertyName] = true;
+        if (typeof Object.isExtensible !== 'function' || (typeof Object.isExtensible === 'function' && Object.isExtensible(xhr))) {
+            xhr[DisabledPropertyName] = true;
```

Comment:
This property is required otherwise this would cause additional telemetry to be re-sent for the same request non-stop and unlike fetch, there is currently no additional alternative to stop this looping...

We took this approach (adding the DisabledPropertyName to the xhr) to avoid having to whitelist every possible URL used to report telemetry (not everyone uses the same/common URL, some people proxy the requests back to their own domain).

Reply:
We also assign the telemetry data against xhr as xhr.ajaxData (line 559 in the _openHandler()), removing this line would effectively make all XHR telemetry disabled...

Which means it's the same as setting the config value "disableAjaxTracking" to true to completely disable all XHR telemetry without any of these code changes. There are also some additional items within InstrumentHook.ts where we store the previous function and hook function calls against the individual methods of XHR.

Do you know if Salesforce platform also locks down window.fetch? As this would have a similar issue, So I'm not sure that this as is simple straight forward solution...

Reply:
@MSNev With the above change, i'm not seeing additional telemetry getting re-sent non-stop.  I see a pageview request.  And then we have a couple custom events that get batched and sent.  That all seems to be working....At least what I can see in the network terminal.  If it was a continual loop, i'd expect to see duplicate calls in the network tab of Chrome.

<img width="705" alt="Screen Shot 2020-03-10 at 10 15 19 AM" src="https://user-images.githubusercontent.com/599418/76334498-bea1ca00-62b8-11ea-9532-ce2ecf12e307.png">

If I run Object.isExtensible(fetch) it returns true;  Is there a way to configure this library to use fetch instead?

Reply:
It's probably not getting resent (looping) because of the other issue below where we populate the xhr.ajaxData -- this is caught internally. But this also means that automatic ajax instrumentation is not going to work either.